server/api-manager.js

```diff
@@ -1007,6 +1032,11 @@ class MindbodyQueries {
               }
               return Promise.resolve(data);
             });
+        })
+        .catch((err) => {
+          if (err.message.includes("ENOTFOUND")) {
+            return Promise.reject(Error("Mindbody failed too many times: " + err.message));
+          }
```

Comment:
I don't understand this. We're applying this catch to the initial (first page) request, and I guess it catches errors that happen during any the requests for subsequent pages. Why does this situation need special handling for ENOTFOUND? The way I see it is that this code block would only receive ENOTFOUND if all 5 retries fail and the final retry received an ENOTFOUND... but in that case, I'm happy handing the ENOTFOUND to the caller. So, I don't think we need any special handling of ENOTFOUND here.
Anyway, when I run this code and get an error that's _not_ ENOTFOUND, I get bad behavior. The makeRequest Promise resolves to the undefined value and returns that as the result of the request (usually getClientVisits in the commission report for me).
I don't know what should be here, but I think this is problematic and we'd be better off just removing the whole catch block.

Reply:
I mostly added this so that we would get a custom message if we hit 5 enotfound errors, eg. we could respond and say something about failing x times. I now see where the bad behavior could come from, and I think that could be eliminated by returning the original error if it isnt enotfound. My reason for doing this is mostly that enotfound is a bad error to pass to the user. If you think this would be better handled elsewhere that can certainly be done by dropping this entirely. I am really open to either so whichever others prefer @DanielBrewerPsu  @roperasher.